client/components/dot-pager/index.jsx

```diff
@@ -83,76 +84,36 @@ export const DotPager = ( {
 	...props
 } ) => {
 	const [ currentPage, setCurrentPage ] = useState( 0 );
-	const [ pagesStyle, setPagesStyle ] = useState();
-	const pagesRef = useRef();
-	const numPages = Children.count( children );
-
-	function useUpdateLayout( enabled, currentPageIndex, updateLayout ) {
-		// save callback to a ref so that it doesn't need to be a dependency of other hooks
-		const savedUpdateLayout = useRef();
-		useEffect( () => {
-			savedUpdateLayout.current = updateLayout;
-		}, [ updateLayout ] );
-
-		// fire when the `currentPageIndex` parameter changes
-		useEffect( () => {
-			if ( ! enabled ) {
-				return;
-			}
-
-			savedUpdateLayout.current();
-		}, [ enabled, currentPageIndex ] );
-
-		// fire when the window resizes
-		useEffect( () => {
-			if ( ! enabled ) {
-				return;
-			}
-
-			const onResize = () => savedUpdateLayout.current();
-			window.addEventListener( 'resize', onResize );
-			return () => window.removeEventListener( 'resize', onResize );
-		}, [ enabled ] );
-	}
-
-	const updateEnabled = hasDynamicHeight && numPages > 1;
 
-	useUpdateLayout( updateEnabled, currentPage, () => {
-		const targetHeight = pagesRef.current?.querySelector( '.is-current' )?.offsetHeight;
-		setPagesStyle( targetHeight ? { height: targetHeight } : undefined );
-	} );
+	const numPages = Children.count( children );
 
 	useEffect( () => {
 		if ( currentPage >= numPages ) {
 			setCurrentPage( numPages - 1 );
 		}
 	}, [ numPages ] );
 
+	const handleSelectPage = ( index ) => {
+		setCurrentPage( index );
+		onPageSelected && onPageSelected( index );
```

Comment:
Just a syntax preference, feel free to ignore

```suggestion
		onPageSelected.?( index );
```

Reply:
THis new syntax didn't work for me.

Reply:
Interesting, would be curious to know what didn't work

Reply:
Me too, actually! In this instance, a function call with optional chaining should work the same way as with the boolean evaluation.

Reply:
This is what I see in the editor. 
<img width="603" alt="Screen Shot 2021-12-02 at 3 28 28 PM" src="https://user-images.githubusercontent.com/115071/144519417-7cb0a492-ff56-4946-a209-03cf5330bf92.png">

Reply:
🤦  My bad, sorry. The correct syntax would be

```suggestion
		onPageSelected?.( index );
```

Reply:
@enejb here's some more information about the optional chaining syntax: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining